### DISCUSSION TITLE
Failed to use in IDBFS when built with WasmFS and FORCE_FILESYSTEM

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#21891] [Q&A]
emcc 3.1.55

Build with those flags: `WasmFS`, `FORCE_FILESYSTEM `is succeed.
but when using in `IDBFS `,emscripten throw an error about missing include to`-lidbfs` .
but build with `-lidbfs`  get an error: "using -lidbfs is not currently supported in WasmFS." (emsdk\upstream\emscripten\src\library_idbfs.js line 316)

**Comment by sbc100** [comment]
I'm not sure wasmfs has an idbfd backend yet.  Currently I don't see 4 backends (at least only 4 headers that end if _backend.h):

```
$ ls system/lib/wasmfs/*_backend.h
system/lib/wasmfs/js_impl_backend.h
system/lib/wasmfs/memory_backend.h
system/lib/wasmfs/pipe_backend.h
system/lib/wasmfs/proxied_async_js_impl_backend.h
```

**Reply by sbc100** [reply]
@tlively can confirm

**Reply by tlively** [reply]
Yes, confirmed. There is no IDB backend for WasmFS yet.

**Comment by aharondavid** [comment]
Meantime, until `indexDB ` will be implemented in `WasmFS`, is there a way to build project with `WasmFS `alongside `-lidbfs`?

**Reply by sbc100** [reply]
No, I believe the old FS and wasmfs (of which `-lidbfs` is part) are mutually exclusive.

**Comment by aharondavid** [comment]
Is latest emscripten version has implementation in IDBFS for WasmFS ?

**Reply by kripken** [reply]
I believe no one has worked on it yet.
